pkg/recipes/terraform/config/config_test.go

```diff
@@ -634,11 +635,20 @@ func Test_AddProviders(t *testing.T) {
 			require.NoError(t, err)
 
 			// assert
-			actualConfig, err := os.ReadFile(getMainConfigFilePath(workingDir))
+			var actualConfig, expectedConfig map[string]any
+
+			actualConfigBytes, err := os.ReadFile(getMainConfigFilePath(workingDir))
 			require.NoError(t, err)
-			expectedConfig, err := os.ReadFile(tc.expectedConfigFile)
+			err = json.Unmarshal(actualConfigBytes, &actualConfig)
 			require.NoError(t, err)
-			require.Equal(t, string(expectedConfig), string(actualConfig))
+
+			expectedConfigBytes, err := os.ReadFile(tc.expectedConfigFile)
+			require.NoError(t, err)
+			err = json.Unmarshal(expectedConfigBytes, &expectedConfig)
+			require.NoError(t, err)
+
+			// This performs a deep comparison of the two maps.
+			require.Equal(t, expectedConfig, actualConfig)
```

Comment:
Converting the actual and the expected configs to maps and doing an equality check.

Reply:
BTW you can also use: https://pkg.go.dev/github.com/stretchr/testify/require#JSONEq